src/dag/write.rs

```diff
@@ -1,55 +1,195 @@
-use super::chunk::Chunk;
 use super::key::Key;
+use super::{chunk::Chunk, meta_generated::meta};
 use super::{read, Result};
 use crate::kv;
+use async_recursion::async_recursion;
+use async_std::sync::RwLock;
+use futures::future::try_join_all;
+use futures::try_join;
+use std::collections::HashSet;
+
+#[derive(Debug, Default)]
+struct HeadChange(Option<String>, Option<String>);
```

Comment:
[taste] I pretty much always regret using tuples for anything. Normal struct?

Reply:
I had it as a struct but `unzip` wants a tuple so I changed it to that to reduce the amount of work needed